enumerable.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    temp = self
+    i = 0
+    while i < temp.length
+      yield(temp[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    temp = self
+    i = 0
+    while i < temp.length
+      yield(temp[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    my_each { |e| temp << e if yield(e) }
+    temp
+  end
+
+  def my_all?
+    test = true
+    my_each { |d| break unless test == yield(d) }
+    test
+  end
+
+  def my_any?
+    test = false
+    my_each { |u| break if test == yield(u) }
+    test
+  end
+
+  def my_none?
+    test = true
+    my_each { |t| break unless test == !yield(t) }
+    test
+  end
```

Comment:
In `my_none?`
- enumerables my_none? returns true if the block never returns true for all elements
- enumerables my_none? when no block or argument is given returns true only if none of the collection members is true
- enumerables my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- enumerables my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- enumerables my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern